datastore/modelmanager.go

```diff
@@ -462,3 +474,21 @@ func (db *DB) checkBoolQuery(row *sql.Row) bool {
 
 	return found
 }
+
+// defaultAPIKey creates a default API key if the field is empty
+func (db *DB) defaultAPIKey(model *Model) error {
```

Comment:
Not sure if I like the name and signature of the method.
I'd prefer this style:
func (db *DB) buildValidOrDefaultAPIKey(originalKey string) (string, err) {..}

and use it like:
model.APIKey, err = buildValidOrDefaultAPIKey(model.APIKey)

In any case, if you want to directly modify model (not recommended style in certain book I read recently and that can become your nightmare), at least modify the method name to indicate that it replaces whitespaces, checks minimum size and generate a new api key (difficult name, but no matter if it is really long)